lib/challenge.rb

```diff
@@ -1,7 +1,25 @@
 # frozen_string_literal: true
 
 class FibonacciRange
+  include Enumerable
+  @previous_fib = 0
+  @current_fib = 1
+
+  def initialize (stop)
+    @stop = stop
+  end
+
+  def each
+    array = [@previous_fib,@current_fib]
+    @next_fib = 0
+    while @next_fib < @stop
+      puts @next_fib = @current_fib + @previous_fib
+      array << @next_fib
+    end
+  end
```

Comment:
Close!

Remember, when we implement an `each` method we have to `yield` each member. In this case, we have to yield each fibonacci number. 🙂 

> The class must provide a method each, which yields successive members of the collection.

From the [`Enumerable docs`](https://ruby-doc.org/core-2.6/Enumerable.html)

```rb
  def each
    current = 0
    successor = 1
    while current < stop
      yield current
      current, successor = successor, current + successor
    end
    self
  end
```